code/drv_ea/src/wattrex_driver_ea/drv_ea.py

```diff
@@ -4,286 +4,326 @@
 
-        self.__properties = DrvEaPropertiesC(model = model, serial_number = serial_number, \
-                                            max_volt_limit = max_voltage_limit, \
-                                            max_current_limit = max_current_limit, \
-                                            max_power_limit = max_power_limit)
-        return self.__properties
 
-
-    def __initialize_control(self) -> None:
-        ''' Enable remote control and turn it off.
+    def __get_nominal_value(self, payload: str) -> int:
+        ''' Get nominal value.
         Args:
-            - None.
+            - payload (str): Payload of the message.
         Returns:
-            - None.
+            - result (int): Nominal value.
         Raises:
             - None.
         '''
-        self.device_handler.send_msg('SYSTem:LOCK: ON')
-        self.device_handler.send_msg('OUTPut: OFF')
-
-
-    def disable(self, channel: int = 1) -> None:
-        ''' Set the device in standby mode.
+        result = 0
+        msg = DrvScpiCmdDataC(data_type = DrvScpiCmdTypeE.WRITE_READ,
+                port = self.__port, payload = payload)
+        self.__tx_chan.send_data(msg)
+        # Wait until receive the message
+        time_init = time()
+        while (time() - time_init) < _MAX_WAIT_TIME:
+            sleep(_TIME_BETWEEN_ATTEMPTS)
+            if not self.__rx_chan.is_empty():
+                command_rec : DrvScpiCmdDataC = self.__rx_chan.receive_data()
+                msg = command_rec.payload[0].split(' ')
+                result = int(float(msg[0]) * _MILI_UNITS)
+        return result
+
+
+    def get_data(self) -> DrvEaDataC:
+        '''Read the device data.
         Args:
-            - None
+            - None.
         Returns:
-            - None
+            - result (DrvEaDataC): Returns the device data.
         Raises:
-            - None 
+            - None.
         '''
-        if (channel<1 or channel>2) or ('2384' not in self.__properties.model and channel != 1):
-            log.error("Try to apply command to a channel doesn´t exist, "+
-                    "will apply to the only available")
-            raise ValueError
-        self.device_handler.send_msg(f'OUTPut OFF (@{channel})')
-        self.__actual_data.mode = DrvEaModeE.STANDBY
-
-
-    def set_cc_mode(self, curr_ref: int, voltage_limit: int, channel: int = 1) -> None:
+        log.info("Get meas...")
+        result: DrvEaDataC = DrvEaDataC(mode = self.last_data.mode, \
+                                        status = DrvBaseStatusC(DrvBaseStatusE.COMM_ERROR), \
+                                        voltage = 0, current = 0, power = 0)
+
+        msg = DrvScpiCmdDataC(data_type = DrvScpiCmdTypeE.WRITE_READ, \
+                            port = self.__port, \
+                            payload = _ScpiCmds.GET_MEAS.value)
+        self.__tx_chan.send_data(msg)
+        time_init = time()
+        while (time() - time_init) < _MAX_WAIT_TIME:

```

Comment:
Active wait is not allowed here. Remove while. Add flags of for receiveing commands

Reply:
Active wait is only done when initilizating device, afterwards will be a flag for getting meas to not saturate scpi